### DISCUSSION TITLE
Tree View - Expand/Collapse States - Properties ?

### DISCUSSION THREAD

**Discussion by markusalbrecht-procam** [AvaloniaUI/Avalonia#13903] [Q&A] [answered]
Currently the TreeView control always collapses in case we use multiple documents and switch between those. Each Document has it's own treedata that is pushed to the TreeViewModel on document change. Is it possible to store the states of the items collapsed and expanded ? Is there a treeview property that forbids the auto collapsing ?

![image](https://github.com/AvaloniaUI/Avalonia/assets/56921919/70fddd56-1149-4dab-addb-60fb68c165b4)

![image](https://github.com/AvaloniaUI/Avalonia/assets/56921919/7bdb7e4a-b12a-43b0-ba02-ef5ba0577f4a)

After switching it is again collapsed - how can we avoid this ?

![image](https://github.com/AvaloniaUI/Avalonia/assets/56921919/9667c4e4-fc8a-4365-9d12-5b0bf1839ddf)

Unfortunately there is no documentation about the available properties for the TreeView control.
https://docs.avaloniaui.net/docs/reference/controls/detailed-reference/treeview-1

**Comment by timunie** [comment]
This is more of a question than an issue to me. So please use Q&A or Telegram https://t.me/Avalonia for questions that are not a bug or feature request. I'll convert it into a Q&A for now.

**Reply by markusalbrecht-procam** [reply]
thanks sorry picked the wrong item

**Comment by timunie** [comment]
You can add a Style for TreeViewItem and bind to IsExpanded afaik. If you use CompiledBindings, you have to set x:DataType on the Style or the Setter.

**Reply by markusalbrecht-procam** [reply]
Ok but this will then wither expand all or collapse all - so in case the user collapses/expands several items we will need to find a way to store this information

**Reply by timunie** [reply]
In your NodeViewModel you would have a TwoWay-Binding. That will store individual state.

**Reply by markusalbrecht-procam** [reply]
Understood, yes we are already thinking of expanding our Node to have the Expanded state as property, this we can then bind and store. I will know more on wednesday - then I will mark the question as done if it works.

**Reply by markusalbrecht-procam** [reply, marked as answer]
We have extended our TreeNode and Save the IsExpanded state there then we update the IsExpanded states of the TreeViewItems on re-load via Styles. This solved the problem. 

Added IsExpanded property to our TreeNode
![image](https://github.com/AvaloniaUI/Avalonia/assets/56921919/b60ab16e-5610-4208-85b8-96262bd72751)

Using it in Styles:
![image](https://github.com/AvaloniaUI/Avalonia/assets/56921919/db375a47-d932-46b1-877e-3a726661b9c8)
